shell/common/asar/archive_win.cc

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Slack Technologies, Inc.
+// Contributors: Weiyun Dai (https://github.com/WeiyunD/), Andrew Lay
+// (https://github.com/guohaolay) Use of this source code is governed by the MIT
+// license that can be found in the LICENSE file.
+
+#include "shell/common/asar/archive.h"
+
+#include <algorithm>
+#include <sstream>
+
+#include "base/logging.h"
+#include "base/strings/string_util.h"
+#include "base/strings/string_util_win.h"
+#include "shell/common/asar/asar_util.h"
+
+using namespace std;
+
+#define INTEGRITY_CHECK_RESOURCE_TYPE L"Integrity"
+#define INTEGRITY_CHECK_RESOURCE_ITEM L"ElectronAsar"
+
+namespace asar {
+
+absl::optional<base::FilePath> Archive::RelativePath() const {
+  // In case user has Windows long path enabled, the exe path should still no
+  // larger than the asar path.
+  size_t buf_len = max(path_.value().size(), (size_t)MAX_PATH);
+  wchar_t* buf = new (std::nothrow) wchar_t[buf_len];
+  if (!buf) {
+    LOG(WARNING) << " " << __func__
```

Comment:
Blocking: Shouldn't this be `FATAL`? If the fuse is enabled and this part of the logic fails we should error on the side of caution and prevent the file read, so somehow this needs to fail, not silent pass.

Reply:
Updated